fhd_utils/interpol_2d.pro

```diff
@@ -22,13 +22,13 @@ mask_i_threshold = dimension/2
 FOR i=0, dimension-1 DO BEGIN
     mask_sum = Total(mask[i, *])
     IF mask_sum LE mask_i_threshold THEN CONTINUE
-    result_x[i, *] = interpol(image_use[i, *], dimension)
+    result_x[i, *] = interpol(image_use[i, *], dimension, /NAN)
```

Comment:
The nan-safe version you are using is also slower. Perhaps add `nan_safe=nan_safe` as a keyword to `interpol_2d` with a default `nan_safe=0`, and pass that in here as `result_x[i, *] = interpol(image_use[i, *], dimension, nan=nan_safe)`. Then you could add that keyword in the places you know need it.

Reply:
I've implemented a `nan_safe` option. I should note that in this code, masking is done via nans. So if a mask is applied, `nan_safe` is turned on automatically.